3.2rc1 test_absolute_times fails on i386

When compiling on i386, the `test_absolute_times` tests fail. They succeed on other 32 bit little endian architectures, so this is probably a x87 issue.
```
__________________________ test_absolute_times[False] __________________________

data = (array([0.00000000e+00, 2.86139335e+00, 2.26851454e+00, 5.51314769e+00,
       7.19468970e+00, 4.23106460e+00, 9.80764...0.00845458, 0.00673573, 0.00502084, 0.00647447]), {'depth': 0.2, 'duration': 0.16, 'period': 2.0, 'transit_time': 0.5})
timedelta = False

    @pytest.mark.parametrize('timedelta', [False, True])
    def test_absolute_times(data, timedelta):
        [...]
        # Check compute_stats
    
        stats1 = bls1.compute_stats(0.2 * u.day, 0.05 * u.day, Time('2019-06-04T12:34:56'))
        stats2 = bls2.compute_stats(0.2 * u.day, 0.05 * u.day, 1 * u.day)
    
        for key in stats1:
            if key == 'transit_times':
>               assert_quantity_allclose((stats1[key] - start).to(u.day), stats2[key])

astropy/timeseries/periodograms/bls/tests/test_bls.py:437: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

actual = <Quantity [1.11022302e-16, 2.00000000e-01, 4.00000000e-01, 6.00000000e-01,
           8.00000000e-01, 1.00000000e+00, ...0000e+00, 9.00000000e+00, 9.20000000e+00, 9.40000000e+00,
           9.60000000e+00, 9.80000000e+00, 1.00000000e+01] d>
desired = <Quantity [ 0. ,  0.2,  0.4,  0.6,  0.8,  1. ,  1.2,  1.4,  1.6,  1.8,
            2. ,  2.2,  2.4,  2.6,  2.8,  3. , ...7. ,  7.2,  7.4,  7.6,  7.8,
            8. ,  8.2,  8.4,  8.6,  8.8,  9. ,  9.2,  9.4,  9.6,  9.8,
           10. ] d>
rtol = 1e-07, atol = None, kwargs = {}
np = <module 'numpy' from '/usr/lib/python3/dist-packages/numpy/__init__.py'>
_unquantify_allclose_arguments = <function _unquantify_allclose_arguments at 0xf56c1df4>

    def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
                                 **kwargs):
        """[...]"""
        import numpy as np
        from astropy.units.quantity import _unquantify_allclose_arguments
        np.testing.assert_allclose(*_unquantify_allclose_arguments(
>           actual, desired, rtol, atol), **kwargs)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatch: 1.96%
E       Max absolute difference: 3.55271368e-15
E       Max relative difference: inf
E        x: array([1.110223e-16, 2.000000e-01, 4.000000e-01, 6.000000e-01,
E              8.000000e-01, 1.000000e+00, 1.200000e+00, 1.400000e+00,
E              1.600000e+00, 1.800000e+00, 2.000000e+00, 2.200000e+00,...
E        y: array([ 0. ,  0.2,  0.4,  0.6,  0.8,  1. ,  1.2,  1.4,  1.6,  1.8,  2. ,
E               2.2,  2.4,  2.6,  2.8,  3. ,  3.2,  3.4,  3.6,  3.8,  4. ,  4.2,
E               4.4,  4.6,  4.8,  5. ,  5.2,  5.4,  5.6,  5.8,  6. ,  6.2,  6.4,...

astropy/tests/helper.py:478: AssertionError
```
Same for `test_absolute_times[True]`.
Looks like one should use only a relative difference here since the first entry is zero.

--- a/astropy/timeseries/periodograms/bls/tests/test_bls.py
+++ b/astropy/timeseries/periodograms/bls/tests/test_bls.py
@@ -434,7 +434,7 @@ def test_absolute_times(data, timedelta):
 
     for key in stats1:
         if key == 'transit_times':
-            assert_quantity_allclose((stats1[key] - start).to(u.day), stats2[key])
+            assert_quantity_allclose((stats1[key] - start).to(u.day), stats2[key], atol=1e-10 * u.day)
         elif key.startswith('depth'):
             for value1, value2 in zip(stats1[key], stats2[key]):
                 assert_quantity_allclose(value1, value2)

